### ISSUE TITLE
Noam scheduler 的 lr

### ISSUE THREAD

**Issue by Chiyu-Song** [thu-coai/CDial-GPT#52]
你好，

发现训练脚本中可能存在的问题：

使用noam scheduler的话initial lr应该设置成1吧，因为noam_lambda的返回值每次会乘以initial lr作为当前step的lr，设置成5e-5似乎会使lr比预期小20000倍。

此外，noam中的step以及其他名字中带有“step“的参数，是否应该以每次“optimizer.step()”为一个单位算作一个step，而不是engine的每个iteration？

如有理解错误之处请见谅。

**Comment by lemon234071** [comment]
您误会了 ， 仔细阅读代码可以发现我们给的inital_lr是针对线性微调的初始lr（--linear_scheduler）， 预训练时的noam lr如论文中所述，通过noam计算公式计算出初始lr， 使得最高峰lr为 文中所报的xe-5数量级的lr，具体请参考论文。至于step， 保证的是 整个训练过程中的lr的曲线变化以控制学习过程，使得训练充分，协同batch size和accum来均衡 训练效率和学习充分度。

**Comment by Chiyu-Song** [comment]
好的，那我没有问题了
